### PR TITLE
[3.2] Backport test contract fixes

### PR DESCRIPTION
**Pull request by oschwaldp-oci**
Resolving build issues with test-contracts

Building test contracts with prior versions of `mandel.cdt` requires the use of a version identifier:

`v1.7.x` requires use of `USE_EOSIO_CDT_1_7_X`
`v1.8.x` requires use of `USE_EOSIO_CDT_1_8_X`
`v3.0.0-rc1` and beyond does not require special parameters.

Build command argument example for building with `mandel.cdt` `v1.7.0`

`cmake -DUSE_EOSIO_CDT_1_7_X=true -DEOSIO_COMPILE_TEST_CONTRACTS=true -DCMAKE_BUILD_TYPE=RelWithDebInfo ..`

Resolves: https://github.com/eosnetworkfoundation/mandel/issues/459

Backports: https://github.com/EOSIO/eos/pull/9114
Backports: https://github.com/EOSIO/eos/pull/9390

### CODE REVIEW

File: unittests/CMakeLists.txt [hunk 1]
```diff
@@ -3,22 +3,26 @@ include(ExternalProject)
...
    set(EOSIO_WASM_OLD_BEHAVIOR "Off")
-   find_package(eosio.cdt REQUIRED)
+   find_package(cdt REQUIRED)
 
-   message( STATUS "Building contracts in directory `eos/unittests/test-contracts/`" )
+   set(CMAKE_ARGS_VAL -DCMAKE_TOOLCHAIN_FILE=${CDT_ROOT}/lib/cmake/cdt/CDTWasmToolchain.cmake -DEOSIO_COMPILE_TEST_CONTRACTS=${EOSIO_COMPILE_TEST_CONTRACTS} )
+   if( USE_EOSIO_CDT_1_7_X)
+      list(APPEND CMAKE_ARGS_VAL -DUSE_EOSIO_CDT_1_7_X=${USE_EOSIO_CDT_1_7_X})
+   endif()
+   message( STATUS "Building contracts in directory `mandel/unittests/test-contracts/`" )
```
Comment (heifner) [inline unittests/CMakeLists.txt:12]: I think replace `mandel/` with `./`

File: unittests/CMakeLists.txt [hunk 2]
```diff
@@ -3,22 +3,26 @@ include(ExternalProject)
...
+      CMAKE_ARGS ${CMAKE_ARGS_VAL}
       UPDATE_COMMAND ""
       PATCH_COMMAND ""
       TEST_COMMAND ""
       INSTALL_COMMAND ""
       BUILD_ALWAYS 1
    )
 else()
-  message( STATUS "Not building contracts in directory `eos/unittests/test-contracts/`" )
+  message( STATUS "Not building contracts in directory `mandel/unittests/test-contracts/`" )
```
Comment (heifner) [inline unittests/CMakeLists.txt:25]: I think replace `mandel/` with `./`

**Review by heifner — CHANGES_REQUESTED** [review]
Did you test with both eosio/cdt 1.7 and mandel/cdt?
Update the PR description with known supported cdt versions.

File: unittests/CMakeLists.txt (hunk 2 above)
Comment (oschwaldp-oci) [reply unittests/CMakeLists.txt]: Good idea.  Addressed here: 77480b98bc4ad325ec59577ca5c3ca6a46a1b71b

File: unittests/CMakeLists.txt (hunk 1 above)
Comment (oschwaldp-oci) [reply unittests/CMakeLists.txt]: Good idea.  Addressed here: 77480b98bc4ad325ec59577ca5c3ca6a46a1b71b

**Review by heifner — APPROVED** [review]

**Pull request merged**